app/src/main/java/org/greenstand/android/TreeTracker/managers/UserLocationManager.kt

```diff
@@ -66,6 +70,10 @@ class UserLocationManager(private val locationManager: LocationManager,
     }
 
     fun stopLocationUpdates() {
+        Timber.d("Request to stop location updates submitted")
+        if (locationUpdateLiveDate.hasObservers())
```

Comment:
This is the main fix I have added to address the location being not captured in the local storage.